frontend/src/components/category-stats.tsx

```diff
@@ -0,0 +1,317 @@
+          
+          refLines.push({
+            x: stat.academic_year,
+            label: changeLabel,
+            color: changeColor,
+            labelPosition: isLastYear ? 'left' : 'right',
+          } as any); // Use 'as any' to bypass TypeScript check since this property works
+        }
+        previousOrganiser = stat.course_organiser;
+      });
+    });
+
+    return {
+      chartData: combinedData,
+      courseCodes: codes,
+      referenceLines: refLines,
+    };
+  }, [stats]);
+
+  // Colors for different course codes
+  const colors = [
+    "var(--mantine-primary-color-6)",
+    "var(--mantine-color-blue-6)",
+    "var(--mantine-color-green-6)",
+    "var(--mantine-color-yellow-6)",
+    "var(--mantine-color-red-6)",
+    "var(--mantine-color-violet-6)",
+    "var(--mantine-color-orange-6)",
+    "var(--mantine-color-teal-6)",
+  ];
+
+  if (loading) {
+    return (
+      <Stack gap="md">
+        <Skeleton height={300} />
+        <Skeleton height={300} />
+      </Stack>
+    );
+  }
+
+  if (error) {
+    return (
+      <Alert color="red" title="Error">
+        Failed to load course statistics: {error.message}
+      </Alert>
+    );
+  }
+
+  if (!stats || stats.length === 0) {
+    return (
+      <Alert color="blue" title="No Data Available">
+        No grade statistics are available for this category. This may be because:
+        <ul>
+          <li>The category doesn't have any associated Euclid codes</li>
+          <li>Course data hasn't been loaded for the associated courses</li>
+          <li>Statistics are not available for the courses in this category</li>
+        </ul>
+      </Alert>
+    );
+  }
+
+  return (
+    <Stack gap="lg">
+      <Box>
+        <Title order={3} mb="md">
+          Course Marks Over Time
+        </Title>
+        <Text size="sm" c="dimmed" mb="md">
+          Hover over data points to see mean marks, standard deviation, and course organiser information.
+          Vertical lines mark course organiser changes.
+        </Text>
+        <Paper withBorder p="md">
+          <LineChart
+            h={400}
+            data={chartData}
+            dataKey="year"
+            series={courseCodes.map((code, index) => ({
+              name: code,
+              color: colors[index % colors.length],
+              strokeWidth: 2,
+            }))}
+            referenceLines={referenceLines}
+            curveType="linear"
+            withLegend
+            withTooltip
+            withDots
+            gridAxis="xy"
+            yAxisProps={{
+              domain: [0, 100],
+              tickFormatter: (value) => `${Math.round(value)}%`,
+            }}
+            tooltipProps={{
+              content: ({ active, payload, label }) => {
+                if (active && payload && payload.length) {
+                  return (
+                    <Paper p="sm" withBorder shadow="md">
+                      <Text size="sm" fw={600} mb="xs">{label}</Text>
```

Comment:
Text here is too small for the "heading" of a tooltip; I suggest md size or even lg, it doesn't particularly take up extra space anyway since it's like 7 characters e.g. "2024-25".